File: bindings/octave/test/test_logger.m
function test_logger()
%TEST_LOGGER  Logger surface over the raw SWIG-generated Octave binding.
%
%   The C ABI's promise is the free functions ptiff_logger_set_level /
%   ptiff_logger_level, and that emitting a log line below/at the current
%   threshold never crashes. Mirrors the Python/Ruby logger tests.

c = ptiff_constants();
orig = ptiff_logger_level();

ptiff_logger_set_level(c.PTIFF_LOG_ERROR);
assert_ptiff(ptiff_logger_level() == c.PTIFF_LOG_ERROR, 'level = error');

% Emitting at a level below the threshold must be safe (and filtered).
ptiff_logger_log(c.PTIFF_LOG_TRACE, 'this trace line is filtered out');
ptiff_logger_log(c.PTIFF_LOG_INFO, 'this info line is filtered out');
ptiff_logger_log(c.PTIFF_LOG_ERROR, 'this error line is emitted');

ptiff_logger_set_level(c.PTIFF_LOG_DEBUG);
assert_ptiff(ptiff_logger_level() == c.PTIFF_LOG_DEBUG, 'level = debug');
ptiff_logger_log(c.PTIFF_LOG_DEBUG, 'backend debugging');
ptiff_logger_log(c.PTIFF_LOG_WARN, 'a warning');

ptiff_logger_set_level(orig);

fprintf('test_logger: OK\n');
end

File: crates/ptiff-python/python/tests/test_logger.py
"""Logger surface (feature parity with SWIG ptiff.logger)."""

import ptiff_pyo3 as ptiff


def test_logger_roundtrip():
    original = ptiff.Logger.level()
    try:
        ptiff.Logger.set_level(ptiff.PTIFF_LOG_ERROR)
        assert ptiff.Logger.level() == ptiff.PTIFF_LOG_ERROR

        # Emitting at/below the threshold must be safe.
        ptiff.Logger.log(ptiff.PTIFF_LOG_TRACE, "filtered trace")
        ptiff.Logger.log(ptiff.PTIFF_LOG_INFO, "filtered info")
        ptiff.Logger.log(ptiff.PTIFF_LOG_ERROR, "emitted error")

        ptiff.Logger.set_level(ptiff.PTIFF_LOG_DEBUG)
        assert ptiff.Logger.level() == ptiff.PTIFF_LOG_DEBUG
        ptiff.Logger.log(ptiff.PTIFF_LOG_WARN, "a warning")
    finally:
        ptiff.Logger.set_level(original)


def test_logger_level_constants_on_class():
    assert ptiff.Logger.TRACE == ptiff.PTIFF_LOG_TRACE
    assert ptiff.Logger.DEBUG == ptiff.PTIFF_LOG_DEBUG
    assert ptiff.Logger.INFO == ptiff.PTIFF_LOG_INFO
    assert ptiff.Logger.WARN == ptiff.PTIFF_LOG_WARN
    assert ptiff.Logger.ERROR == ptiff.PTIFF_LOG_ERROR
    assert ptiff.Logger.CRITICAL == ptiff.PTIFF_LOG_CRITICAL
    assert ptiff.Logger.OFF == ptiff.PTIFF_LOG_OFF
